sydent/util/emailutils.py

```diff
@@ -64,16 +64,24 @@ def sendEmail(
     )
 
     allSubstitutions = {}
-    for k, v in substitutions.items():
-        allSubstitutions[k] = v
-        allSubstitutions[k + "_forhtml"] = escape(v)
-        allSubstitutions[k + "_forurl"] = urllib.parse.quote(v)
 
-    # We add randomize the multipart boundary to stop user input from
-    # conflicting with it.
-    allSubstitutions["multipart_boundary"] = generateAlphanumericTokenOfLength(32)
+    # use jinja for rendering if jinja templates are present
+    if templateFile.endswith(".j2"):
+        # We add randomize the multipart boundary to stop user input from
+        # conflicting with it.
+        substitutions["multipart_boundary"] = generateAlphanumericTokenOfLength(32)
+        template = sydent.template_environment.get_template(templateFile)
+        mailString = template.render(substitutions)
+    else:
+        with open(templateFile) as template_file:
+            for k, v in substitutions.items():
+                allSubstitutions[k + "_forhtml"] = escape(v)
+                allSubstitutions[k + "_forurl"] = urllib.parse.quote(v)
+            allSubstitutions["multipart_boundary"] = generateAlphanumericTokenOfLength(
+                32
+            )
+            mailString = template_file.read() % allSubstitutions
```

Comment:
Since `allSubstitutions` is now only used in this branch, we can move it into here. It is also usually best to keep the file open for as short as possible! (I think I gave you the previous code, my bad!)

```suggestion
        allSubstitutions = {}
        for k, v in substitutions.items():
            allSubstitutions[k + "_forhtml"] = escape(v)
            allSubstitutions[k + "_forurl"] = urllib.parse.quote(v)
        allSubstitutions["multipart_boundary"] = generateAlphanumericTokenOfLength(
            32
        )
        with open(templateFile) as template_file:
            mailString = template_file.read() % allSubstitutions
```